### ISSUE TITLE
regression with v0.15.0: `lncli closedchannels` gives "unable to fetch chan info: unexpected EOF" and "unable to fetch chan info: invalid public key: unsupported format: c7"

### ISSUE THREAD

**Issue by C-Otto** [lightningnetwork/lnd#6548]
### Background

I've been experimenting on testnet and used several versions of the `master` branch to run my two nodes. I think I started with v0.14.3, but I can't remember - I can dig in deeper, if that helps.

On both nodes I can't run `lncli closedchannels` (same issue via gRPC). Note that everything else seems to work fine!

On the first node I get:

```
unable to fetch chan info: unexpected EOF
```

On the second node I get:

```
unable to fetch chan info: invalid public key: unsupported format: c7
```
(where I think the public key is related to gRPC / TLS / ...)

### Your environment

* version of `lnd`: recent master up to v0.15.0-beta.rc-1
* which operating system (`uname -a` on *Nix): Linux Linux home 5.16.2 #<i></i>1 SMP PREEMPT Thu Jan 20 21:00:54 CET 2022 x86_64 GNU/Linux
* version of `btcd`, `bitcoind`, or other backend: bitcoind v23
* any other relevant environment details: testnet, Go 1.18beta1

### Steps to reproduce

Start lnd on testnet, update from v0.14.3 (I think) to several master commits from the past couple of days. Run `lncli closedchannels`.

### Expected behaviour

Closed channels are shown.

### Actual behaviour

Error messages, see above.

**Comment by Roasbeef** [comment]
Updated an older testnet node, it was able to start up properly, but hit an issue when running `lncli closedchannels`, I get a similar error:
```
[lncli] rpc error: code = Unknown desc = unable to fetch chan info: invalid public key: unsupported format: 10
```

**Comment by Roasbeef** [comment]
Likely error source: https://github.com/lightningnetwork/lnd/blob/0ec88b534678f713bcdc06c87610b9184df64207/channeldb/channel.go#L1431, but things don't quite match up since `lncli closedchannels` doesn't appear to hit that code path.

**Comment by Roasbeef** [comment]
Tracked things to https://github.com/lightningnetwork/lnd/blob/c647d32cf7d3da17e3f94f937e1e5cf6b29e2b2f/rpcserver.go#L4296, which then attempts to access the historical channel bucket, where this error is sourced from.

**Comment by Roasbeef** [comment]
Found the culprit: https://github.com/lightningnetwork/lnd/pull/6347/files#r875295125. 

So an incomplete migration. Can be resolved my filling in that missing gap, though we may want to also change that field to be a TLV in the process? cc @yyforyongyu

**Comment by Roasbeef** [comment]
Remedy path: https://github.com/lightningnetwork/lnd/pull/6347/files#r875296783 

Any mainnet nodes affected by this _should_ be able to re-run that migration (can add a one off flag). Or we can implement the fix as another migration, which could then also do that LTV thing I mentioned.

**Comment by Roasbeef** [comment]
Confirmed that this PR fixes the issue: https://github.com/lightningnetwork/lnd/pull/6551

**Comment by C-Otto** [comment]
I merged #6551 and it's running fine on my node. Thanks a lot!

**Comment by Roasbeef** [comment]
Fixed by #6551
